app/assets/javascripts/discourse-markdown-it/src/features/watched-words.js

```diff
@@ -55,35 +50,27 @@ export function setup(helper) {
   helper.registerPlugin((md) => {
     const matchers = [];
 
-    if (md.options.discourse.watchedWordsReplace) {
-      Object.entries(md.options.discourse.watchedWordsReplace).forEach(
-        ([regexpString, options]) => {
-          const word = toWatchedWord({ [regexpString]: options });
-
-          matchers.push({
-            word: new RegExp(options.regexp, options.case_sensitive ? "" : "i"),
-            pattern: createWatchedWordRegExp(word),
-            replacement: options.replacement,
-            link: false,
-          });
-        }
-      );
-    }
+    md.options.discourse.watchedWordsReplace?.forEach(
+      ([regexp, caseSensitive, , replacement]) => {
```

Comment:
"caseSensitive, , replacement" looks like a typo. 

I also liked the readability of named options for case sensitivity and replacements a little better.

Reply:
see the other comment, but also - I think I might be able to get rid of that extra (`word`) property. I don't think I seen any uses of it.